src/java_bytecode/expr2java.cpp

```diff
@@ -491,9 +483,10 @@ std::string expr2javat::convert(
   const exprt &src,
   unsigned &precedence)
 {
+  const typet &type=ns.follow(src.type());
   if(src.id()=="java-this")
```

Comment:
@kroening what's the rationale for "java-this" (I note that there also is "cpp-this")?